AudiusProject/audius-protocol
# [CON-506] Migrate files with custom storage paths

### Description
- Runs a background task to find files with custom storagePaths and move them to a standard path. Gated by env var `migrateFilesWithCustomStoragePath` (default true) and tracked by prom metric `files_migrated_from_custom_path`
- Removes disk deletion from migrations because automated disk deletion already takes care of it, and it sometimes errors with "file not found" when automated disk deletion finds a file first
- Speeds up legacy migration (batch size 1000->3000) and fixes bug where it sorts in the reverse direction
- Makes legacy migration overwrite existing files and `touch` the file first to prevent a race condition where disk pruning sees the copied file (with outdated `mtime`) and prunes it before it's updated in the db
- Fixes infinite loop that affected legacy migrations for directories only


### Tests

1. Set `backgroundDiskCleanupCheckEnabled` and `migrateFilesWithCustomStoragePath` to false in config.js
2. `A seed clear; A seed create-user; A seed upload-track`
3. In db: `UPDATE "Files" SET "storagePath" = CONCAT('/home/customStorage', SUBSTRING("storagePath" FROM 20)) WHERE true;`
4. `docker exec -it cn1_creator-node_1 /bin/sh` and move each file manually: `mkdir -p /home/customStorage; cd /file_storage/files; ls` then for each dir: `mv <dir>/<CID> /home/customStorage`
5. Set `migrateFilesWithCustomStoragePath` to true in config.js
6. After a minute, storagePath in the db should be updated to what it was before. In the container, `/file_storage/files` should have all the files
7. Set `backgroundDiskCleanupCheckEnabled` to true in config.js and set `DAYS_BEFORE_PRUNING_ORPHANED_CONTENT` to 0 in diskManager.ts. In the container, `/home/customStorage` should have only empty folders
8. Redo this test, but for step 4 delete the files instead of moving them: `rm -rf /file_storage/files`

In these tests I also verified that legacy storage path migration was not finding any of the custom file paths.


### Monitoring - How will this change be monitored? Are there sufficient logs / alerts?
- Check for any occurrences of "Error fixing fileRecord"
- We should eventually see every node's value stop increasing for the metric `files_migrated_from_custom_path`
- The following query should decrease to eventually return 0: `SELECT COUNT(*) FROM "Files" WHERE "storagePath" NOT LIKE '/file_storage/%'`
